gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDataset.java

```diff
@@ -137,13 +139,17 @@ Collection<CopyEntity> generateCopyEntities(FileSystem targetFs, CopyConfigurati
       // TODO: determine whether unnecessarily expensive to repeatedly re-create what should be the same FS: could it
       // instead be created once and reused thereafter?
       FileSystem actualSourceFs = getSourceFileSystemFromFileStatus(srcFileStatus, defaultHadoopConfiguration);
+      Path toPath = PathUtils.getRootPathParent(srcPath);
```

Comment:
I'm unsure if we need to recursively ensure that all permissions are the same until root, I think this can cause issues cross cluster. At least with hive copy, we only validate the table path and its parent (not saying that this is the most robust way of doing it though, we've seen issues with that approach too).

I think the safest and most predictable approach is to ensure that all the directories that Gobblin will create will have the same permissions as the source.